web/viewer.js

```diff
@@ -506,41 +510,59 @@ var PDFViewerApplication = {
       self.progress(progressData.loaded / progressData.total);
     }
 
-    PDFJS.getDocument(parameters, pdfDataRangeTransport, passwordNeeded,
-                      getDocumentProgress).then(
-      function getDocumentCallback(pdfDocument) {
-        self.load(pdfDocument, scale);
-        self.loading = false;
-      },
-      function getDocumentError(exception) {
-        var message = exception && exception.message;
-        var loadingErrorMessage = mozL10n.get('loading_error', null,
-          'An error occurred while loading the PDF.');
-
-        if (exception instanceof PDFJS.InvalidPDFException) {
-          // change error message also for other builds
-          loadingErrorMessage = mozL10n.get('invalid_file_error', null,
-                                            'Invalid or corrupted PDF file.');
-        } else if (exception instanceof PDFJS.MissingPDFException) {
-          // special message for missing PDF's
-          loadingErrorMessage = mozL10n.get('missing_file_error', null,
-                                            'Missing PDF file.');
-        } else if (exception instanceof PDFJS.UnexpectedResponseException) {
-          loadingErrorMessage = mozL10n.get('unexpected_response_error', null,
-                                            'Unexpected server response.');
-        }
+    function getDocumentCallback(pdfDocument) {
+      self.load(pdfDocument, scale);
+      self.loading = false;
+    }
+
+    function getDocumentError(exception) {
+      var message = exception && exception.message;
+      var loadingErrorMessage = mozL10n.get('loading_error', null,
+        'An error occurred while loading the PDF.');
+
+      if (exception instanceof PDFJS.CancelGetDocumentException) {
+        loadingErrorMessage = mozL10n.get('cancel_get_document_error', null,
```

Comment:
I don't think we need to show any errors to user -- first, it was his decision in first place and, second, it may confuse the user if he chooses to load a different document and he received an error instead.